backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/pporder/process/WEBUI_PP_Order_Pick_HU.java

```diff
@@ -0,0 +1,290 @@
+	private OrderLineId getSalesOrderLineId()
+	{
+		final IView view = getView();
+		if (view instanceof PPOrderLinesView)
+		{
+			final PPOrderLinesView ppOrderLinesView = PPOrderLinesView.cast(view);
+			return ppOrderLinesView.getSalesOrderLineId();
+		}
+		else
+		{
+			return null;
+		}
+	}
+
+	@ProcessParamLookupValuesProvider(//
+			parameterName = WEBUI_M_HU_Pick_ParametersFiller.PARAM_M_PickingSlot_ID, //
+			dependsOn = WEBUI_M_HU_Pick_ParametersFiller.PARAM_M_ShipmentSchedule_ID, //
+			numericKey = true, //
+			lookupSource = DocumentLayoutElementFieldDescriptor.LookupSource.lookup)
+	private LookupValuesList getPickingSlotValues(final LookupDataSourceContext context)
+	{
+		final WEBUI_M_HU_Pick_ParametersFiller filler = WEBUI_M_HU_Pick_ParametersFiller
+				.pickingSlotFillerBuilder()
+				.shipmentScheduleId(shipmentScheduleId)
+				.build();
+
+		return filler.getPickingSlotValues(context);
+	}
+
+
+
+	@Override
+	protected ProcessPreconditionsResolution checkPreconditionsApplicable()
+	{
+		final ProcessPreconditionsResolution eligibleView = checkEligibleView();
+		if (!eligibleView.isAccepted())
+		{
+			return eligibleView;
+		}
+
+		if (!getView().getPlanningStatus().isComplete())
+		{
+			return ProcessPreconditionsResolution.rejectWithInternalReason("PPOrder is not complete");
+		}
+
+		final Set<HuId> distinctHuIds = retrieveSelectedHuIds();
+		if (distinctHuIds.isEmpty())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@Override
+	@RunOutOfTrx
+	protected String doIt()
+	{
+		final Set<HuId> distinctHuIds = retrieveSelectedHuIds();
+
+		// TODO refine OR  implement a best matching HUs strategy (based on received Qty)
+
+		for (final HuId huId : distinctHuIds)
+		{
+			// get shipment schedule & compare Quantity Ordered / Picked (stock UOM) ; based on same UOM
+			final I_M_ShipmentSchedule shipmentSchedule = shipmentScheduleBL.getById(shipmentScheduleId);
+			final BigDecimal qtyOrdered = shipmentSchedule.getQtyToDeliver();
+			final BigDecimal qtyPicked = shipmentSchedule.getQtyPickList();
+			if(qtyPicked.compareTo(qtyOrdered) < 0)
+			{
+				pickHU(huId);
+			}
+		}
+
+		// invalidate view in order to be refreshed
+		getView().invalidateAll();
+
+		return MSG_OK;
+	}
+
+	private void pickHU(final HuId huId)
+	{
+		pickingCandidateService.pickHU(PickRequest.builder()
+											   .shipmentScheduleId(shipmentScheduleId)
+											   .pickFrom(PickFrom.ofHuId(huId))
+											   .pickingSlotId(pickingSlotId)
+											   .build());
+		// NOTE: we are not moving the HU to shipment schedule's locator.
+		pickingCandidateService.processForHUIds(ImmutableSet.of(huId),
+												shipmentScheduleId,
+												OnOverDelivery.ofTakeWholeHUFlag(isTakeWholeHU),
+												getView().getPpOrderId());
+	}
+
+	private Set<HuId> retrieveSelectedHuIds()
+	{
+		final DocumentIdsSelection selectedRowIds = getSelectedRowIds();
```

Comment:
please extract the code that is common with WEBUI_PP_Order_PrintLabel into a super-class, utility-class or elsewhere, to deduplicate it

Reply:
PS: also please look into doing the same for the code that is common with WEBUI_M_HU_Pick

Reply:
>> ok, so I guess WEBUI_M_HU_Pick will stay in its own right, jsut won't be accessible from the production modal, right?
Yes indeed